frontend/src/components/game/Console.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { SubmissionResult, TestCase } from '../../api/Problem';
+import { BoldText } from '../core/Text';
+import { ConsoleTextArea } from '../core/Input';
+import { SmallButton } from '../core/Button';
+
+const Content = styled.div`
+  height: 100%;
+`;
+
+const FixedContent = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+type ConsoleProps = {
+  testCases: TestCase[],
+  submission: SubmissionResult | null,
+  onRun: (input: string) => void,
+};
+
+// This function refreshes the width of Monaco editor upon change in container size
+function Console(props: ConsoleProps) {
+  const [input, setInput] = useState('');
+  const [output, setOutput] = useState('');
+
+  const { testCases, submission, onRun } = props;
+
+  useEffect(() => {
+    setInput(testCases?.length ? testCases[0].input : '');
+  }, [testCases]);
+
+  useEffect(() => {
+    setOutput(submission ? submission.output : '');
+  }, [submission]);
+
+  return (
+    <Content>
+      <FixedContent>
+        <SmallButton onClick={() => onRun(input)}>Run Code</SmallButton>
+      </FixedContent>
+      <div>
+        <BoldText>Input</BoldText>
+        <ConsoleTextArea value={input} onChange={(e) => setInput(e.target.value)} />
+      </div>
+      <div>
+        <BoldText>Output</BoldText>
+        <ConsoleTextArea value={output} readOnly />
+      </div>
```

Comment:
Is this UI issue fixable? [Screencast here](https://www.loom.com/share/c1de1a7b4fa849289b480dc8ba4d8867), image below.

<img width="1279" alt="UI Issue with Draggable Input" src="https://user-images.githubusercontent.com/7987279/97759126-9bcd5b80-1abd-11eb-8301-43e5497bf997.png">

If not, maybe you could temporarily disable the draggability of the text boxes, or just leave it as-is -- not a huge issue.

Reply:
Good catch - I made the console scrollable and added constraints to the text box sizes, so it should no longer allow for this situation